packages/mobile/src/screens/sign-on-screen/screens/SelectGenreScreen.tsx

```diff
@@ -1,98 +1,104 @@
+    const valueIndex = values.genres.indexOf(value)
+
+    if (valueIndex > -1) {
+      // Already checked
+      newValues.genres.splice(valueIndex, 1)
+    } else {
+      // Unchecked
+      newValues.genres.push(value)
+    }
+    setValues(newValues)
+  }
+
+  return (
+    <ScrollView testID='genreScrollView'>
+      <Flex gap='s' direction='row' wrap='wrap'>
+        {genres.map((genre) => (
+          <SelectablePill
+            label={genre.label}
+            key={genre.value}
+            onPress={() => handleChange(genre.value)}
+            isSelected={values.genres.includes(genre.value)}
+            size='large'
+          />
+        ))}
+      </Flex>
+    </ScrollView>
+  )
+}
 
 export const SelectGenreScreen = () => {
-  const { value: displayName } = useSelector(getNameField)
-  const { value: handle } = useSelector(getHandleField)
   const dispatch = useDispatch()
   const navigation = useNavigation<SignUpScreenParamList>()
 
   const handleSubmit = useCallback(
     (values: SelectGenreValues) => {
-      const genres = Object.keys(values).filter((genre) => values[genre])
+      const genres = values.genres
       dispatch(setField('genres', genres))
       navigation.navigate('SelectArtists')
     },
     [dispatch, navigation]
   )
 
   return (
-    <ScrollView testID='genreScrollView'>
-      <View>
-        <Text>{displayName}</Text>
-        <Text>{handle}</Text>
-      </View>
-      <View>
-        <Text>{messages.header}</Text>
-        <Text>{messages.description}</Text>
-      </View>
-      <Formik initialValues={initialValues} onSubmit={handleSubmit}>
-        {({ values, setValues, handleSubmit }) => (
-          <View>
-            {genres.map((genre) => {
-              const { label, value } = genre
-              const checked = !!values[value]
-
-              const handleChange = () => {
-                setValues({
-                  ...values,
-                  [value]: !checked
-                })
-              }
-
-              return (
-                <Pressable
-                  key={value}
-                  testID={label}
-                  accessibilityRole='checkbox'
-                  accessibilityState={{ checked }}
-                  accessibilityLiveRegion='polite'
-                  onPress={handleChange}
-                  style={{ backgroundColor: checked ? 'purple' : undefined }}
-                >
-                  <Text>{label}</Text>
-                </Pressable>
-              )
-            })}
-            <Button title={messages.continue} onPress={() => handleSubmit()} />
-          </View>
-        )}
-      </Formik>
-    </ScrollView>
+    <Formik
+      initialValues={initialValues}
+      onSubmit={handleSubmit}
+      validationSchema={toFormikValidationSchema(selectGenresSchema)}
+      validateOnBlur
+      validateOnChange
+    >
+      {({ handleSubmit: triggerSubmit, dirty, isValid }) => (
+        <View>
+          <ReadOnlyAccountHeader />
```

Comment:
so we wouldnt want to make this a child of page and have no top padding?

Reply:
hmm I didnt consider that given the page comes with padding but in hindsight that's probably better than doing the dimensions - 196 thing

Reply:
yeah ok I like that better